cloud/terraform/gcloud_config_builder.py

```diff
@@ -0,0 +1,135 @@
+from cloud.terraform.base_config_builder import BaseConfigBuilder
+
+
+class GCloudConfigBuilder(BaseConfigBuilder):
+    cloud_name = 'gcloud'
+    cloud_provider_definition = {'google': {'source': 'hashicorp/google', 'version': '~> 3.5.0'}}
+
+    default_ssh_user = 'user'
+
+    ssh_enabled_tag = 'ssh-enabled'
+
+    def __init__(self, resources_dict, ssh_key_path):
+        super().__init__(resources_dict)
+
+        self.ssh_key_path = ssh_key_path
+        self.project = resources_dict['project']
+
+    def build_providers(self):
+        all_regions = self.__get_all_regions_from_resources_file()
+        for region in all_regions:
+            self.providers_tf['provider'][self.cloud_providers[self.cloud_name]]\
+                .append(self.__new_gcloud_provider(self.project, region))
+
+        return self.providers_tf
+
+    def __get_all_regions_from_resources_file(self):
+        instances_regions = [i['region'] for i in self.resources_dict['instances']]
+
+        return list(dict.fromkeys(instances_regions))
+
+    def __new_gcloud_provider(self, project, region):
+        zone = f'{region}-c'
+
+        return {
+            'project': project,
+            'region': region,
+            'zone': zone,
+        }
+
+    def build_resources(self):
+        self.resources_tf['resource']['google_compute_network'] = {}
+        self.resources_tf['resource']['google_compute_firewall'] = {}
+        self.resources_tf['resource']['google_compute_instance'] = {}
+
+        network_name = self.create_resource_name(['vpc'])
+
+        self.__new_gcloud_network(network_name)
+        self.__new_gcloud_firewall_rule(network_name)
+
+        for instance in self.resources_dict['instances']:
+            instance['google_compute_network'] = network_name
+            self.__new_gcloud_instance(instance)
+
+        return self.resources_tf
+
+    def __new_gcloud_firewall_rule(self, network_name):
+        name = self.create_resource_name(['firewall-rule'])
+
+        allow_rule = {
+            'protocol': 'tcp',
+            'ports': ['22'],
+        }
+
+        new_rule = {
+            'name': name,
+            'network': network_name,
+            'target_tags': [self.ssh_enabled_tag],
+            'source_ranges': ['0.0.0.0/0'],
+            'allow': allow_rule,
+            'depends_on': [
+                'google_compute_network.{}'.format(network_name)
+            ],
+        }
+
+        self.resources_tf['resource']['google_compute_firewall'][name] = new_rule
+
+    def __new_gcloud_network(self, network_name):
+        new_vpc = {
+            'name': network_name,
+            'auto_create_subnetworks': True
+        }
+
+        self.resources_tf['resource']['google_compute_network'][network_name] = new_vpc
+
+    def __new_gcloud_instance(self, instance):
+        if not instance['instance_type']:
+            instance['instance_type'] = 'e2-micro'
```

Comment:
Are we sure this instance type is enough for our testing? Maybe we should add a TODO so we don't forget to verify it

Reply:
No, not sure really, but we can also specify the instance type in the resources.json. The same for AZ and AWS. so far, this default instance type has worked well, although only 2 images have been tested. I think we will discover if this is an issue in the future.

Reply:
Cool, let's keep this in mind then